lib/segment/src/data_types/groups.rs

```diff
@@ -82,3 +98,77 @@ impl GroupId {
         }
     }
 }
+
+/// A value that can be used as a temporary ID
+#[derive(Debug, Serialize, Deserialize, JsonSchema, Eq, PartialEq, Clone, Hash)]
+#[serde(untagged)]
+pub enum PseudoId {
+    String(String),
+    NumberU64(u64),
+    NumberI64(i64),
+}
+
+impl Display for PseudoId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PseudoId::String(s) => write!(f, "{}", s),
+            PseudoId::NumberU64(n) => write!(f, "{}", n),
+            PseudoId::NumberI64(n) => write!(f, "{}", n),
+        }
+    }
+}
+
+impl From<GroupId> for PseudoId {
+    fn from(id: GroupId) -> Self {
+        match id {
+            GroupId::String(s) => Self::String(s),
+            GroupId::NumberU64(n) => Self::NumberU64(n),
+            GroupId::NumberI64(n) => Self::NumberI64(n),
+        }
+    }
+}
+
+impl TryFrom<PseudoId> for PointIdType {
+    type Error = ();
```

Comment:
Is it possible to unify the error with something else than unit and then discard the error at the call site?

Reply:
Good point, done